### PR TITLE
feat(JSR-1216): multi-chat support

### PR DESCRIPTION
**Pull request by eebasadre20**
## Description

Please explain what kind of changes you made in this PR.

## Issues
- https://jitera.atlassian.net/browse/JSR-1216

## Videos

Video to prove your work

## Test

What kind of tests were performed and what were the results?

# Checklist

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation to reflect the change
- [ ] I have updated the README.md to reflect the change
- [ ] I have added unit tests to prove that the update is effective and the feature works

### CODE REVIEW

File: ruby/lib/payloads/v2/asset_pb.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+# Generated by the protocol buffer compiler.  DO NOT EDIT!
+# source: v2/asset.proto
+
+require 'google/protobuf'
+
+
+descriptor_data = "\n\x0ev2/asset.proto\x12\tschema.v2\"\x1f\n\x05\x41sset\x12\x16\n\x0e\x62\x61se64_content\x18\x01 \x01(\tb\x06proto3"
+
+pool = Google::Protobuf::DescriptorPool.generated_pool
```
Comment (comictvn) [inline ruby/lib/payloads/v2/asset_pb.rb:10]: where asset file defined?

File: ruby/lib/payloads/v2/session_pb.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+# Generated by the protocol buffer compiler.  DO NOT EDIT!
+# source: v2/session.proto
+
+require 'google/protobuf'
+
+
+descriptor_data = "\n\x10v2/session.proto\x12\tschema.v2\"\x1f\n\x07Session\x12\x14\n\x0c\x61\x63\x63\x65ss_token\x18\x01 \x01(\tb\x06proto3"
+
+pool = Google::Protobuf::DescriptorPool.generated_pool
```
Comment (comictvn) [inline ruby/lib/payloads/v2/session_pb.rb:10]: where session file defined?

File: ruby/lib/payloads/v2/asset_pb.rb (hunk 1 above)
Comment (eebasadre20) [reply ruby/lib/payloads/v2/asset_pb.rb]: @comictvn forgot to clean up my previous generate protobuf.

File: ruby/lib/payloads/v2/session_pb.rb (hunk 2 above)
Comment (eebasadre20) [reply ruby/lib/payloads/v2/session_pb.rb]: https://github.com/Jitera-Product/jitera_proto/pull/146#discussion_r1713173836

File: protos/v2/chat.proto [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  }
+}
+
+message Reference {
+  optional string name = 1;
+  optional string origin = 2;
+  optional ReferenceType reference_type = 3;
+  optional string category = 4;
+  optional string content = 5;
+  map<string, string> metadata = 6;
```
Comment (zrg-team) [inline protos/v2/chat.proto:91]: Struct instead for more flexibility

File: protos/v2/chat.proto [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  }
+}
+
+message Resource {
+  optional string name = 1;
+  optional string content = 2;
+  optional ResourceType resource_type = 3;
+  optional Status status = 4;
+  optional string category = 5;
+  map<string, string> metadata = 6;
```
Comment (zrg-team) [inline protos/v2/chat.proto:68]: Struct

File: protos/v2/chat.proto [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+message Message {
+  int32 project_id = 1;
+  optional Status status = 2;
+  optional MessageAssistant assistant = 3;
+  optional MessageFrom from = 4;
+  optional string content = 5;
+  optional bool hidden = 6;
+  map<string, string> metadata = 7;
```
Comment (zrg-team) [inline protos/v2/chat.proto:31]: Struct

File: protos/v2/chat.proto [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+  repeated MessageAsset assets = 8;
+  repeated Resource resources = 9;
+  repeated Reference references = 10;
+  google.protobuf.Timestamp created_at = 11;
```
Comment (zrg-team) [inline protos/v2/chat.proto:35]: Number would be ok ?

**Review by zrg-team — CHANGES_REQUESTED** [review]

**Review by zrg-team — APPROVED** [review]

**Pull request merged**